src/IssueViz.Security/IssueVizSecurityPackage.cs

```diff
@@ -41,7 +42,9 @@ namespace SonarLint.VisualStudio.IssueVisualization.Security
     [Guid("D7D54E08-45E1-49A6-AA53-AF1CFAA6EBDC")]
     [ProvideMenuResource("Menus.ctmenu", 1)]
     [ProvideToolWindow(typeof(HotspotsToolWindow), MultiInstances = false, Transient = true, Style = VsDockStyle.Tabbed, Window = VsWindowKindErrorList, Width = 700, Height = 250)]
-    [ProvideToolWindow(typeof(TaintToolWindow), MultiInstances = false, Transient = true, Style = VsDockStyle.Tabbed, Window = VsWindowKindErrorList, Width = 700, Height = 250)]
+    [ProvideToolWindow(typeof(TaintToolWindow), MultiInstances = false, Transient = false, // Note: Transient must false when using ProvideToolWindowVisibility
+        Style = VsDockStyle.Tabbed, Window = VsWindowKindErrorList, Width = 700, Height = 250)]
+    [ProvideToolWindowVisibility(typeof(TaintToolWindow), TaintIssuesExistUIContext.GuidString)]
```

Comment:
When we made this change in secondary locations panel, it introduced a problem with the persistence that we had to fix using the frame close:
https://github.com/SonarSource/sonarlint-visualstudio/commit/e0e78e54041c67ff3260573922f39d06d4947c28#diff-9f42bc19311338b66ca2e7dd10a3d6fd54f56ae33f85544277fd064861e59e04

Is that needed?
What happens if you force-open the tool window and close VS? will it disappear next time VS launches? (it didn't for secondary locations so we had to force close the frame)

Reply:
Good catch!

Reply:
FYI I added this code, but it didn't seem to stop the window being reopened next time VS started. I'll push this code as it is.
I have the "close frame" code plus tests in a separate commit and will test it again tomorrow.

Reply:
Hmm, have you tried to reset the hive? Just to make sure that it's not caching something improperly.